### PR TITLE
Nav api alr

### PR DESCRIPTION
**Pull request by reid-andrew**
### Type of change made:
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Styling - no new features
### Why is this change required? What problem does it solve?
- This PR adds in functionality to call the Charity Navigator API & return a list of charities to the front end application.
### Were there any challenges that arose while implementing this feature? If so, how were they addressed?
- Unfamiliarity with consuming an API & creating objects out of it's data in Python/Flask. Involved lots of dream driving & stepping through code using a Pry-like tool to see what we were working with at every step & a decent bit of refactoring & research into Python methods.

### CODE REVIEW

File: application/popos/charity.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Charity:
+    def __init__(self, id, name, url, rating):
+        self.id = id
+        self.url = url
+        self.name = name
+        self.rating = rating
```
Comment (reid-andrew) [inline application/popos/charity.py:6]: Popo to create objects returned to FE with data from Charity Navigator API.

File: tests/test_charities.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+import unittest
+import json
```
Comment (reid-andrew) [inline tests/test_charities.py:2]: I know, I know. 

Need to do further research into mocks/stubs in Python/Flask to write API tests.

File: application/controllers/charities.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    type=str,
+    required=False,
+    location=['json']
+)
+
+class CharityResources(Resource):
+    def get(self, search_term=None):
+        search_term  = "" if search_term == None else search_term
+        chars = return_charities(search_term)
+
```
Comment (StephanieFriend) [inline application/controllers/charities.py:48]: Nice thinking ahead so we don't get a null value in the params!

File: application/services/charity_service.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+                'rated': 'TRUE',
+                'minRating': 3,
+                'sort': 'RATING:DESC',
+                'search': search_term
+               }
+
+    response = requests.get(
+                            'https://api.data.charitynavigator.org/v2/organizations',
+                            params=payload
+                            )
```
Comment (StephanieFriend) [inline application/services/charity_service.py:37]: Oooo I LOVE this break out of the service and popo files! Shows a clean line of single responsibility.

File: tests/test_charities.py (hunk 2 above)
Comment (StephanieFriend) [reply tests/test_charities.py:2]: If we have time we'll come back to the testing. It's a little bit harder than the other tests we wrote. Even if it's later down the road we can figure it out. :)

**Review by StephanieFriend — APPROVED** [review]
This looks great! I really like how you split the POPO out into a POPO and a Service. Very clean and easy to follow. I realize there is no testing for this, but we can get a test going even if it is later down the road.

File: application/services/charity_service.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
+from application.popos.charity import Charity
+import requests
```
Comment (kmcgrevey) [inline application/services/charity_service.py:2]: `requests` acts like `Faraday` in Rails

File: application/services/charity_service.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+from dotenv import load_dotenv
+load_dotenv()
+
+def return_charities(search_term):
+    charities = charity_objects(charity_list(search_term))
+
+    return charities
+
+def charity_objects(response):
+    parsed = response.json()
```
Comment (kmcgrevey) [inline application/services/charity_service.py:13]: `json` built-in method with `requests`

**Review by kmcgrevey — APPROVED** [review]

**Pull request merged**